app/src/main/java/org/hackillinois/android/view/admin/ToolsFragment.kt

```diff
@@ -0,0 +1,171 @@
+package org.hackillinois.android.view.admin
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import kotlinx.android.synthetic.main.fragment_admin_events.*
+import kotlinx.android.synthetic.main.fragment_admin_events.view.*
+import kotlinx.android.synthetic.main.fragment_admin_notifications.*
+import kotlinx.android.synthetic.main.fragment_admin_notifications.view.*
+import kotlinx.android.synthetic.main.fragment_admin_stats.*
+import kotlinx.android.synthetic.main.fragment_admin_stats.view.*
+import org.hackillinois.android.App
+import org.hackillinois.android.R
+import org.hackillinois.android.model.*
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import kotlin.concurrent.thread
+
+class ToolsFragment : Fragment() {
+    private val ARG_SECTION_NUM = "section_number"
+
+    companion object {
+        fun newInstance(sectionNumber: Int): ToolsFragment {
+            val fragment = ToolsFragment()
+            val args = Bundle()
+
+            args.putInt(ToolsFragment().ARG_SECTION_NUM, sectionNumber)
+            fragment.arguments = args
+
+            return fragment
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val view: View?
+
+        val sectionNumber = if (arguments == null) 0 else arguments!!.getInt(ARG_SECTION_NUM)
+
+        when (sectionNumber) {
+            0 -> view = createStatsView(inflater, container, savedInstanceState)
+            1 -> view = createEventsView(inflater, container, savedInstanceState)
+            else -> view = createNotifcationsView(inflater, container, savedInstanceState)
+        }
+
+        return view
+    }
+
+    fun createStatsView(inflater: LayoutInflater, container: ViewGroup?,
+                        savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_admin_stats, container, false)
+
+        view.queryBtn.setOnClickListener {
+            thread {
+                val response = App.getAPI().stats.execute()
```

Comment:
I'm not a huge fan of having API calls in Fragments. I would prefer if we could put them into a ViewModel and have LiveData subscribers like the rest of the Fragments. If you take a look at the ScheduleFragment code, Kelly did this with a ViewPager.

Reply:
One potential issue with this is that the toasts you have below will show up again on an orientation change or anything that causes LiveData to refire. Here's a way around that: [link](https://medium.com/androiddevelopers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150)

Reply:
The general look of the ViewModel should be something like:

```Kotlin
class AdminViewModel : ViewModel() {
    val queryResult = MutableLiveData<String>()
    
    fun queryPressed() {
        App.getAPI().stats.enqueue(object : Callback {
            ...
            fun onSuccessOrWhatever() {
                 //parseResult...
                 queryResult.postValue(result)
            } 
        } 
    }
}
```

Reply:
Then in the fragment we can observe these changes which will allow them to be persisted across orientation change and all sorts of lifecycle changes.